### DISCUSSION TITLE
How to use state in socket handler?

### DISCUSSION THREAD

**Discussion by trgsv** [Totodore/socketioxide#67] [Q&A] [answered]
How can I use Axums extract::State inside a socket handler as it only takes one argument (`Arc<Socket<LocalAdapter>>`)?

**Comment by Totodore** [comment, marked as answer]
You can't use axum state extractor with your handler because socketioxide is not an axum handler it is a middleware _in front of axum_ (or anything based on hyper. Currently the lib lacks of some state management. 

The only things you can use are [Extensions](https://docs.rs/socketioxide/latest/socketioxide/struct.Socket.html#structfield.extensions) with a cloned `Arc` of your state for each socket. 

Or to clone an `Arc` of your state in each of your handler.
